### PR TITLE
[block] collect luksDump for all encrypted devices

### PR DESCRIPTION
**Pull request by pmoravec**
Call "cryptsetup luksDump /dev/sd*" for all encrypted devices

Resolves: #1390

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

**Comment by pmoravec** [comment]
I am not sure if `block` plugin is the best place to put the code into, but havent found a better one and creating an extra plugin just for this small enhancement seems a bit overkill for me.

File: sos/plugins/block.py [hunk 1]
```diff
@@ -51,4 +51,15 @@ def setup(self):
                     "fdisk -l %s" % disk_path
                 ])
 
+        lsblk_result = self.call_ext_prog("lsblk -P -o +FSTYPE -p")
```
Comment (bmr-cymru) [inline sos/plugins/block.py]: Perhaps we should also add `lsblk -f``... I don't think we're capturing FSTYPE anywhere.

Comment (bmr-cymru) [reply sos/plugins/block.py]: (we could also do it here - the code isn't really using `-P` name=value parsing at all, so a simple `in` test of the line would work fine. Then we can store the result in the report for analysis use as well).

**Comment by pmoravec** [comment]
Rebased with:

- calling `lsblk -f -a`
- parsing it output and calling `cryptsetup luksDump` for encrypted devices

Since `lsblk` is newly collected while also utilized for generating list of encrypted devices, I suggest to extend `block` plugin.

**Comment by pmoravec** [comment]
`dmsetup` does not collect inactive devices :( So currently it seems to me that `lsblk` is the best option to detect all LUKS devices, and the code should be added either to:

- `block` plugin due to `lsblk` - but the encrypted devices dont have too much common with block devices (this is the state of the PR now)
- to a separate plugin that will collect (or just call?) a command that should belong to `block` plugin

No strong opinion from me here.

**Pull request closed without merging**